src/main/java/be/seeseemelk/mockbukkit/block/data/BlockDataMock.java

```diff
@@ -201,23 +211,65 @@ public boolean equals(Object obj)
 	{
 		Preconditions.checkNotNull(material, "Material cannot be null");
 		// Special Cases
+		BlockDataMock mock = attemptMockByPaperMaterialTags(material);
+		if (mock != null)
+		{
+			return mock;
+		}
+
+		mock = attemptMockByTag(material);
+		if (mock != null)
+		{
+			return mock;
+		}
+
+		return switch (material)
+				{
+					case AMETHYST_CLUSTER -> new AmethystClusterMock(material);
+					default -> new BlockDataMock(material);
+				};
+	}
+
+	/**
+	 * Attempts to construct a BlockDataMock object by matching against Paper MaterialTags. Returns null if the given
+	 * material does not match any supported MaterialSetTag.
+	 *
+	 * @param material Material which we will attempt to mock
+	 * @return BlockDataMock if matched, null otherwise
+	 */
+	static BlockDataMock attemptMockByPaperMaterialTags(@NotNull Material material)
```

Comment:
Can this be private? I don't see a need for anything to ever access this outside of the mock method.

Reply:
Good point and yes I can change that

Reply:
Addressed in my last commit

Reply:
Are you sure? They're still both package-private on the web viewer. Please don't force push unless you have to, it's nice to have a proper git history, and it makes things like this easier because I could've just looked at that specific commit.

Reply:
Clearly not and I must have messed up resetting the commit. _Now_ it should be added.

And sorry about the force-push; I thought it would make the commit history for the changes more compact and less "noisy", but I didn't realize it would make reviewing it on your end more difficult. I will avoid that in the future.